### ISSUE TITLE
Bug: Library not found for Arch64

### ISSUE THREAD

**Issue by FaultyxDx** [SpaiR/imgui-java#165]
### Version

1.86.9

### What happened?

java.lang.UnsatisfiedLinkError: dlopen failed: "/data/app/~~aPFq1K5IGZtPglACKE1Dzw==/xyz.akatski.imguijni-G8FAKsF_TBczARNP_Fv6Hw==/lib/arm64/libimgui-java64.so" is for EM_X86_64 (62) instead of EM_AARCH64 (183)

### Reproduction

i just loaded the sample code is provided in repo and compiled dependency in my application

**Comment by SpaiR** [comment]
Unfortunately, binding doesn't provide builds for arm/arch architectures.

**Comment by ds58** [comment]
I would really like to have arm64 builds for macOS since that is a fairly common platform out there now.

**Comment by phraktle** [comment]
I would also prefer to use this natively. In the meantime, you can install an x86 JVM which will run with Rosetta.

**Comment by kles4enko** [comment]
+1

**Comment by phraktle** [comment]
some projects started using zig for easy cross-compilation of C code, eg. https://github.com/lmdbjava/lmdbjava/pull/217 (though not sure this works here, given the C++ bits, but may be interesting to look into)

**Comment by LawrenceB5477** [comment]
+1

**Comment by fahdwhgads** [comment]
#190 should solve this, but for anyone who wants a quick and dirty solution for Apple Silicon:

1. Clone this repository
2. Go to `buildSrc/src/main/groovy/tool/generator/GenerateLibs.groovy`
3. Apply the following changes so you can target arm64:
```diff
- def mac64 = BuildTarget.newDefaultTarget(BuildTarget.TargetOs.MacOsX, true)
+ def mac64 = BuildTarget.newDefaultTarget(BuildTarget.TargetOs.MacOsX, true, true)
```
```diff
- BuildExecutor.executeAnt(jniDir + '/build-macosx64.xml', commonParams)
+ BuildExecutor.executeAnt(jniDir + '/build-macosxarm64.xml', commonParams)
```
4. From the repository root build a native library for arm64: (you might have to install ant)
```sh
./gradlew imgui-binding:generateLibs -Denvs=macos -Dfreeetype=true -Dlocal
```
6. You'll find the native library under `imgui-binding/build/libsNative/macosxarm64/libimgui-javaarm64.dylib`
7. When starting your application pass the following JVM options: 
```
-Dimgui.library.path=./path/to/library/folder -Dimgui.library.name=libimgui-javaarm64.dylib
```
9. Don't forget to also add the LWJGL native librariess for arm64 to your classpath

**Comment by mcpkdotdev** [comment]
Does anyone know how to solve for Asahi linux running in linux arm64?

**Comment by SpaiR** [comment]
Resolved #223
aarch64 support will be available in the next release
